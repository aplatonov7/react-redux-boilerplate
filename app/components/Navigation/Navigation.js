import React, { PropTypes } from 'react'
import { IndexLink, Link } from 'react-router'
import s from './Navigation.scss'

const Navigation = () => (
  <nav className={s.root}>
    <IndexLink className={s.link} activeClassName={s.active} to="/">Home</IndexLink>
    <Link className={s.link} activeClassName={s.active} to="counter">Counter</Link>
  </nav>
)

Navigation.propTypes = {

}

export default Navigation
