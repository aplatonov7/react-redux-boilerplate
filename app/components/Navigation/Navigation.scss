@import 'variables';
@import 'mixins';

.root {
  display: block;
  margin-bottom: 2rem;
  font-size: 1.5rem;
}

.link {
  color: $color-link;
  text-decoration: none;
  margin: 0 1rem;
  transition: .2s ease-out color;

  &:hover {
    border-bottom: 1px solid currentColor;
  }
}

.active {
  color: darken($color-link, 10%);
}
