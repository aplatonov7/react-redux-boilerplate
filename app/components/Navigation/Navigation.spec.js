import React from 'react'
import renderer from 'react-test-renderer'
import Navigation from './'

describe('Navigation component', () => {
  const item = renderer.create(<Navigation />)

  it('should match snapshot', () => {
    expect(item.toJSON()).toMatchSnapshot()
  })
})
