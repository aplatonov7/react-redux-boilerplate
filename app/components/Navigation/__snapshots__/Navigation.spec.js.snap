exports[`Navigation component should match snapshot 1`] = `
<nav
  className="root">
  <a
    className="link"
    onClick={[Function]}
    style={Object {}}>
    Home
  </a>
  <a
    className="link"
    onClick={[Function]}
    style={Object {}}>
    Counter
  </a>
</nav>
`;
